File: epicerie-vrac-projet-angular/src/app/models/panier.ts
import { jsonIgnore } from "json-ignore";
import { Achat } from "./achat";
import { Client } from "./client";

export class Panier extends Achat {

    constructor(client: Client) {
        super();
        this.client = client;
    }

    // @jsonIgnore()
    public client: Client;
}

File: epicerie-vrac-projet/src/main/java/fr/formation/epicerievracprojet/models/Panier.java
package fr.formation.epicerievracprojet.models;

import javax.persistence.Entity;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@Getter
@Setter
@ToString(callSuper = true)
@Entity
public class Panier extends Achat {
	
	@ToString.Exclude
	@JsonIgnore
	@OneToOne
	private Client client;
	
}
